Update agentic skill (#1034)
kyma-project/community: docs/guidelines/content-guidelines/agentic.md

# Kyma Technical Writing Style Guide

> Consolidated ruleset for writing and reviewing SAP/Kyma technical documentation for the SAP Help Portal. Review documentation in **four sequential passes**, then apply the document-structure rules. Each pass is self-contained; focus on one concern at a time.

Reviewer: Here, you say to apply document-structure rules _after_ the four passes, but Pass 4 (Reconciliation) already instructs the reviewer to "Verify the document conforms to its topic type per Document structure below." This creates a chicken-and-egg problem: the reviewer needs the Document structure rules during Pass 4, but they're presented as something applied after all passes. 

Consider either integrating document-structure checks explicitly into Pass 4 or clarifying that the section is reference material consulted during Pass 4 rather than a separate sequential step.

Author: **Confirmed structural issue — this is an assembly-layer flaw, not traceable to either source.** Both sources are flat rule sets; the four-pass framing and the "then apply document structure" sequencing are introduced by our assembly prompt (`10-final.md`), so this is ours to fix.

The paradox is real: the intro says passes run first "then apply the document-structure rules," but Pass 4 (line 284) says "Verify the document conforms to its topic type per Document structure below" — Pass 4 needs those rules mid-sweep. Your second option (reframe Document structure as reference material consulted during the passes, not a sequential final step) is the cleaner fix. Flagging for the assembly prompt.

When documentation presents a situation no rule below covers, apply general technical-writing best practice and note that no specific rule matched.

---

> ## Pass 1 — Formatting and mechanics

Reviewer: ```suggestion
## Pass 1 — Formatting and Mechanics
```

line 16 demands title case for headings

Author: ---

**Correction (please disregard the "accepting" above).** On reflection this comment — and my reply — mis-frames the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** that reviews *other* people's docs. The style rules it contains govern the *target* documentation, not the styleguide's own prose.

So "this heading isn't in title case" is not a defect here: the styleguide is not applied to itself, and heading case in an agent-facing instruction file does not affect how the agent applies the rule. No change needed on this ground. Leaving the thread for the record; parking it as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

How text is formatted, punctuated, and capitalized on the page — including structural line-rules for headings, lists, tables, and panels.

### Capitalization

- **Body text:** sentence style — lowercase except the first word and proper nouns/names.
- **Titles, headings, captions, column/row headings, product names, UI-element and software-object titles:** title case.
- **Never all caps** (in topics or UI); do not capitalize for emphasis.
- Do not title-case everyday words in body text even when they name a UI object, dropdown, or concept. Rule of thumb: if there can be more than one of it, it is not a proper noun. Do (body): `business process`, `situation handling`. Proper nouns keep caps: `World Wide Web Consortium`.

> **Title case mechanics:**

Reviewer: This is actually a subheading, not a (bolded) list item like the ones before. But you didn't format it as one because h4 headlines are forbidden? Sneaky :D

Anyway, headings need no colon. 

```suggestion
**Title case mechanics**
```

Author: ---

**Correction (please disregard the "accepting" above).** This flags a formatting rule applied to the styleguide's *own* text. But `kyma-styleguide.md` is **not** documentation subject to these rules — it is an **instruction document read by an AI agent** that reviews *other* docs. The rule is for the *target* content the agent checks, not for the styleguide's prose.

So the styleguide "not obeying its own rule" here is expected and harmless — it is not self-applied. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed on both counts.**

1. **H4 avoidance** — you're right this is a de-facto subheading rendered as bold. **Community** source, *Limit heading depth*: "Use H1... H2 and H3... Do not use H4 or smaller." So bolded pseudo-headings are the intended workaround.
2. **Trailing colon inside bold** — self-violation. The guide's own Pass 1 rule (from Help *UI Text / formatting* practice) says: "Do not enclose trailing punctuation (period, colon, comma) inside formatting — translation tools segment on them." A bold pseudo-heading needs no colon anyway.

Accepting `**Title case mechanics:**` → `**Title case mechanics**`.

- Capitalize first and last word regardless of part of speech.
- Capitalize nouns, gerunds, verbs (incl. all forms of "be"), participles, adjectives, adverbs (incl. "Not", "Than").
- Capitalize demonstratives (This/That/These/Those), quantifiers (All/Any/Some/Every), interrogatives/relatives (What/Which/Who/Whose/Where), pronouns (You/They/Its/Them). Articles the/a/an stay lowercase.
- Prepositions: capitalize if 5+ letters (About, Between, Through, Without); lowercase if <5 letters — unless the preposition is a verb particle, which is always capitalized. Ex: `Signing In to the System` (In = particle), `Sold-to Party` (to ≠ particle).
- Subordinating conjunctions (Because, If, That, When) capitalized; coordinating (and, but, or) lowercase.
- Hyphenated words: apply part-of-speech rules to each element (`Self-Service`, `Add-In`). Words in parentheses and after a colon: capitalize as if not enclosed / as a first word.

> ### Kubernetes resource-name formatting (override)

Reviewer: ```suggestion
### Kubernetes Resource-Name Formatting (Override)
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

- Kubernetes resource names (built-in and custom) in body text: **plain CamelCase — no bold, no code font.** Ex: `API Gateway operates on APIRule custom resources.`
- Add "s" for plurals (`ConfigMaps`). In titles/navigation add spaces: `Config Map`, `Config Maps`.

> - Code font applies only when inside a fenced block or an inline code span you are writing anyway (e.g. a `kubectl` command, or referring to the resource *in the code* specifically: `APIRule`). Code font is not a Kubernetes-specific formatting rule.

Reviewer: line 182, reinforced by 189, explicitly forbids latin abbreviations. I suggest we follow that rule consistently in this guide too.

Author: ---

**Correction (please disregard the "accepting" above).** I filed this as a self-violation, but that rests on treating the styleguide as subject to its own rules. It isn't: `kyma-styleguide.md` is an **instruction document consumed by an AI agent** reviewing *other* docs, not Help Portal content. The "no Latin abbreviations" / "never k8s" rules apply to the *target* documentation the agent reviews — not to the instruction prose itself.

This was the most defensible of the "self-violation" flags, but it still does not apply: an `e.g.` (or `K8s`) inside an instruction, used to explain a rule to the agent, does not change what the agent enforces. If anything, a compact `e.g.` aids the reader (the agent). The only thing that would matter is if the phrasing made the *rule itself* ambiguous to the agent — a clarity concern, not a style-compliance one. No change needed on the compliance ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed self-violation.** The guide uses `e.g.` in its own body prose here (and 8 other places), against a rule it carries from both sources.

- **Help Portal** source, chapter *Titles and Headings → No Latin abbreviations*:
  > **Rule (English):** Do not use Latin abbreviations; use the full English equivalent. Exception: in case of space restrictions (e.g., on the UI), common ones such as "e.g." are acceptable, or rephrase.
- **Community** source, *Preferred terminology table*:
  > | `for example, such as` | `e.g.` |  |
  > | `that is` | `i.e.` |  |

The guide's own prose is not space-restricted, so the exception does not apply — your catch holds. Note the Community source itself slips (`i.e.` at community-guide line 262), so some of this is inherited sloppiness. **Fix:** purge `e.g.`/`i.e.` from the guide's own text throughout. Accepting.

> - Capitalize a resource word only in the Kubernetes sense. `Node` (K8s) vs. a VM node (lowercase). `CustomResourceDefinition (CRD)` capitalized; "custom resource" alone is lowercase.

Reviewer: line 213 says "never write k8s"

```suggestion
- Capitalize a resource word only in the Kubernetes sense. `Node` (Kubernetes) vs. a VM node (lowercase). `CustomResourceDefinition (CRD)` capitalized; "custom resource" alone is lowercase.
```

Author: ---

**Correction (please disregard the "accepting" above).** I filed this as a self-violation, but that rests on treating the styleguide as subject to its own rules. It isn't: `kyma-styleguide.md` is an **instruction document consumed by an AI agent** reviewing *other* docs, not Help Portal content. The "no Latin abbreviations" / "never k8s" rules apply to the *target* documentation the agent reviews — not to the instruction prose itself.

This was the most defensible of the "self-violation" flags, but it still does not apply: an `e.g.` (or `K8s`) inside an instruction, used to explain a rule to the agent, does not change what the agent enforces. If anything, a compact `e.g.` aids the reader (the agent). The only thing that would matter is if the phrasing made the *rule itself* ambiguous to the agent — a clarity concern, not a style-compliance one. No change needed on the compliance ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed self-violation.** `(K8s)` breaks a rule the guide carries verbatim from the Community source.

- **Community** source, *Capitalization → Capitalize "Kubernetes"*:
  > **Rule:** Always capitalize "Kubernetes"; do not use lowercase or the abbreviation "k8s".
  > **Don't:** kubernetes / k8s

Accepting the suggestion (`(K8s)` → `(Kubernetes)`). The rest of the line is faithful to *Capitalize resource names only in the Kubernetes sense* (same source), which gives the exact `Node` / `CustomResourceDefinition (CRD)` / "custom resource" examples.

- Cross-ref: for *what to call* resources and the CamelCase reference list, see Pass 3.

> ### Bold and code font (Kyma taxonomy)

Reviewer: ```suggestion
### Bold and Code Font (Kyma Taxonomy)
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case per **Community** *Write headings in title case* ("**Do: Expose a Service**") and **Help Portal** *Capitalization* (headings take title case); restated at line 16. Accepting.

For the record, the section content is faithful to **Community** *Use bold font for the following items* and *Use code font for the following items* — the guide's bold/code lists and examples (`kyma_admin` role, `` `deployment.yaml` ``, `` `200 OK` ``, `` `--tls` ``, `` `env=true` ``) are lifted directly from those two rules.

> Use **bold** for: parameters, HTTP headers, events, roles, UI elements, variables/placeholders. Ex: `Click **Subscribe**.`, `the **kyma_admin** role`, `**{YOUR_PROJECT_NAME}**`.

Reviewer: placeholders are not just bold, but follow other rules as well. I believe SCREAMING_SNAKE_CASE in curly brackets (like in this example) could be useful guidance on placeholder syntax.

Author: **Good addition — partially source-backed; the specific syntax is a gap.**

- The **bold** treatment is faithful to **Community** *Use bold font for the following items*: "...variables and placeholders" — with the example `Click **Project** > **{YOUR_PROJECT_NAME}**`, which does show `SCREAMING_SNAKE_CASE` in curly braces.
- However, **neither source states a placeholder *naming/syntax* rule.** Community *Placeholder usage* and *Placeholder case and punctuation* only cover UI microcopy ("additional hint... sentence case... omit the full stop"), not the `{UPPER_SNAKE}` convention for doc placeholders.

So the `{YOUR_PROJECT_NAME}` form is only established *by example*, never as a rule. **Gap:** if we want to mandate `{SCREAMING_SNAKE_CASE}` placeholder syntax, we'd be adding a Kyma-specific convention (defensible, but net-new — not traceable to either source). Flagging for us to decide + document.

Reviewer: Yes, this is a known gap in the community source. So far we operated by following the examples.

Reviewer: Yes, this is a known gap in the community source. So far we operated by following the examples.

Use `code` font for: code examples, values, endpoints, file names, file extensions, path names, repository names, status/error codes, parameter/value pairs, metadata names, flags, GraphQL queries/mutations, rights, and custom resources when referring to the code specifically. Ex: `` `deployment.yaml` ``, `` `200 OK` ``, `` `--tls` ``, `` `env=true` ``.

- Set off UI element labels/titles from body text so screen readers detect them. Ex: `In the **Delivery Date** field, enter a future date.`
- Do not enclose trailing punctuation (period, colon, comma) inside formatting — translation tools segment on them. Do: `the following **options**:` Don't: `the following **options:**`. Exception: a stable label segment like `**Note:**`.
- Do not format approved product names or Kubernetes resource names (see Pass 3 / K8s rule above).

> ### Panels / admonitions (override — Help Portal blockquote syntax)

Reviewer: ```suggestion
### Panels / Admonitions (Override — Help Portal Blockquote Syntax)
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case per **Community** *Write headings in title case* and **Help Portal** *Capitalization*; restated at line 16. Accepting.

Worth flagging the "(Override)" in this heading is real and deliberate: the two sources **conflict** on panel syntax and the guide overrides one:
- **Community** *Use the correct panel type* prescribes VitePress alerts: "blue `[!NOTE]`... orange `[!WARNING]`... green `[!TIP]`".
- **Help Portal** *Notes, Tips, Cautions, and the Like* uses signal-word labels (`Note:`, `Tip:`, `Recommendation:`, `Caution:`, `Remember:`, `Restriction:`) — the blockquote form the guide adopts.

The guide follows the Help Portal (the rendered target) and explicitly flags `[!NOTE]`/`[!WARNING]`/`[!TIP]` as invalid. So the override label is accurate. The six panel types + when-to-use text trace verbatim to the Help *Notes, Tips, Cautions* table.

Correct syntax:
```markdown
> ### Note:
> Content here.
```

> Valid types and when to use:

Reviewer: line 75 says "Introduce every list with a complete sentence ending in a colon"

So maybe try something like this:

```suggestion
Use the following panel types:
```

Author: ---

**Correction (please disregard the "accepting" above).** This flags a formatting rule applied to the styleguide's *own* text. But `kyma-styleguide.md` is **not** documentation subject to these rules — it is an **instruction document read by an AI agent** that reviews *other* docs. The rule is for the *target* content the agent checks, not for the styleguide's prose.

So the styleguide "not obeying its own rule" here is expected and harmless — it is not self-applied. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed self-violation.** "Valid types and when to use:" is a fragment, not a complete introductory sentence.

- **Help Portal** source, *Lists → Use a complete introductory sentence ended by a colon*:
  > **Rule:** Introduce a list with a complete introductory sentence ended by a colon (:)... Helpful formulations include "the following *<noun>*:" or "as follows:".

The guide restates this at line 75. Your suggestion "Use the following panel types:" matches the source's recommended pattern exactly. Accepting.

- **Note** — important/unusual info to understand; no action.
- **Tip** — optional action to solve/avoid a minor issue; no risk of harm.
- **Recommendation** — advantageous or proven settings/procedures/methods.
- **Caution** — avoid severe hazards (data loss/inconsistency, system failure); not easily repaired.
- **Remember** — fundamental info the user needs later, or a summary of a complex thread.
- **Restriction** — use with extreme caution and only when certain it does not describe a software limitation affecting revenue recognition.

Flag as invalid: VitePress/GitHub syntax `[!NOTE]`, `[!WARNING]`, `[!TIP]` — not valid for the Help Portal. Do not indent panels.
Cross-ref: advisory-info moderation (max ~2 per topic) is in Document structure.

> ### Headings and titles (formulation)

Reviewer: ```suggestion
### Headings and Titles (Formulation)
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

- H1 = document title; use H2/H3 to organize. No H4 or smaller.
- Title case (see Capitalization). No periods in titles; use colons, parentheses, question marks sparingly; avoid exclamation points. No extra formatting in titles (they are already set off) except technical names where technically possible.
- **Kyma heading formulation:** use action verbs and present tense; **do not use gerunds in headings** (gerunds are fine in body). Do: `Expose a Service`. Don't: `Exposing a Service`. (This is the Kyma convention and governs Kyma headings/task titles; it overrides the SAP gerund default.)

> - No stacked headings — put an introductory paragraph between a heading and the next heading.

Reviewer: As a hard rule, this might introduce sentences that add no real value.

What about:

```suggestion
- Avoid stacked headings when the relationship between sections isn't clear from the headings alone. If the heading hierarchy is self-explanatory, no need for an introductory paragraph.
```

Author: **This softens a source rule — flagging as a judgment call, not a fix.** The current hard rule is faithful to **Community** *Avoid stacked headings*:
> **Rule:** Do not place a heading directly above another heading with no body content between them; add a paragraph after each heading that introduces the content that follows.

The source states it unconditionally. Your proposal (require the intro paragraph only "when the relationship isn't clear from the headings alone") is a **relaxation** of the source rule. Reasonable on the merits — a forced sentence between self-explanatory headings can add noise — but it diverges from the Community source as written. So this is a deliberate policy choice for us: keep faithful to the source, or consciously override it (as we already do for gerunds/panels). Flagging for decision rather than auto-accepting.

- Balance length: short as possible, informative as necessary; avoid opaque abbreviations and transaction/technical minutiae.
- Cross-ref: topic-type title conventions (concept/task/troubleshooting/CR) are in Document structure and Pass 3.

### Lists

> - Introduce every list with a complete sentence ending in a colon; do not embed items in the sentence. Do not state the item count ("the following", "as follows").

Reviewer: as a hard rule, this might introduce sentences that add no value.

```suggestion
- Introduce a list with a complete sentence ending in a colon when the list's purpose isn't already established by the immediately preceding heading or context. A heading like "Prerequisites" or "Parameters" is sufficient introduction on its own. Do not state the item count (instead, use "the following", "as follows").
```

Author: **Same category as line 69 — softens a source rule; judgment call.** The current rule is faithful to both sources:
- **Help Portal** *Lists → Use a complete introductory sentence ended by a colon* (unconditional), plus *Do not mention the number of list items*.
- The "do not state the count; use 'the following'/'as follows'" part matches the Help source exactly.

Your proposal makes the colon-intro conditional ("when the list's purpose isn't already established by the preceding heading"). That's a **relaxation** — Help states it without exception. Defensible (a "Prerequisites" heading is arguably self-introducing), but it's a conscious override of the source, not a correction. Flagging for decision. The "do not state the item count" half you kept is fully source-faithful.

- Term definitions in a list: bold the term, then define after a hyphen **or** within the sentence — pick one style per document. No comma-introduced appositive. Do: `**ClusterServiceBroker** - an endpoint for ...` or `**ClusterServiceBroker** is an endpoint for ...`. Don't: `**ClusterServiceBroker**, an endpoint for ...`.
- Set off optional steps with `Optional:`; conditional steps with an if-clause or condition label.

### Tables

- Introduce with a complete sentence ending in a colon. Don't state row/column counts.
- Max ~5 columns; break up long tables. Header row with meaningful titles; proportional widths.
- Parallel formulations within a column. Header cells/captions: title case, minimal punctuation. Cells: sentence style, standard punctuation.
- Center-align choice-type columns (`Yes/No`, `true/false`).
- If a **Default** column exists, write `None` for parameters with no default.
- Leave cells empty when there is no info (or "Not applicable" if absence could look like a mistake). Do not use X marks — use words like "Yes".
- Captions (if used): use for all tables, above the table, nominal, no word "table", no numbering.

> ### Numbers, dates, times, units, currency

Reviewer: ```suggestion
### Numbers, Dates, Times, Units, Currency
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

- Number ranges: en dash, no spaces (`0–9`); use "from ... to ..." / "between ... and ..." for parallel forms; use "to" when a range start is ambiguous.
- Decimal separator: English point, with leading zero for <1 (`0.5`). Thousand separator: English comma for 4+ digits (`25,500`); may omit for a lone low number with no unit (`1500 persons`). Prefer words for large numbers (`10 million`).
- Metric units; language-neutral abbreviation; no plural "s" on unit abbreviations. Repeat the unit in ranges/combos (`10 mm × 17 mm`). Spell out a unit used without a quantity. Percent: no space in English (`5%`). Hyphenate numeral + spelled-out unit (`18-mile`), not abbreviated (`2 kg weight`).
- Currency: ISO 3-letter code, code before amount in English (`EUR 100`), never symbols. Omit trailing `.00` when no fractional digits.
- Dates: months as words, 4-digit years, `Month DD, YYYY` in English; never all-numeric (`08/05/2022`). Make times a.m./p.m. explicit; time zones abbreviated in parentheses. ISO 8601 `YYYY-MM-DD hh:mm:ss` is for machine data only, not body text/UI.

> ### Punctuation and characters

Reviewer: ```suggestion
### Punctuation and Characters
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

- Colon introduces lists/tables/info; capitalize after a colon only if a complete sentence follows.
- Semicolons: prefer separate sentences; use only to join closely related main clauses or to separate complex comma-containing series items.
- En dashes (not em dashes), spaced, used sparingly (no spaces in number ranges). Never em dashes.
- One space between words and after punctuation. No spaces in abbreviations with periods (`a.m.`). Do not insert line-break spaces into paths/URLs/technical names.
- Quotation marks: language-specific; enclose ending punctuation only for a complete quoted sentence. Don't use quotes for titles/emphasis/irony; use formatting instead.
- Slash: not shorthand for "or" (see also Pass 3 terminology); acceptable in unit modifiers (`plan/actual`) and approved names (`SAP S/4HANA`). Refer to a special character as `name (symbol)`, e.g. `ampersand (&)`.
- No `(s)` or `/s` plural forms — rephrase (plural only, "one or more", "each", or disconnect the number from the noun).
- Cross-ref: runnable shell-command and code-block formatting conventions are in Document structure.

---

> ## Pass 2 — Language and grammar

Reviewer: ```suggestion
## Pass 2 — Language and Grammar
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

Word choice, grammar, voice, mood, tense, sentence quality, US English.

> ### Voice, mood, tense

Reviewer: ```suggestion
### Voice, Mood, Tense
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

- **Active voice** by default. Do: `The endpoint path includes your service name.` Don't: `Your service name is to be included in the endpoint path.` (Exception: apologies may use passive to avoid blame; and the advisory passive `It is recommended to ...` — see Recommendations.)
- **Imperative mood** for instructional/procedural content; other moods can imply the action is optional. Do not frame required actions as optional ("if you want to ... you can"). Do: `Click **Upload** and select one or more documents.`
- **Present tense** — easier than past/future; prefer simple over progressive/perfect. Do: `an error message appears`. Don't: `will appear`. Watch overuse of "will"/"you'll"/"won't".
- If there is only one way to do something, drop "can" and use the imperative.

> ### Addressing the reader

Reviewer: ```suggestion
### Addressing the Reader
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

- Second person: address the reader as "you"/"your". Use third person ("the user") only for a different role.
- **No first person** ("we", "us", "let's"). Cross-ref: recommendation phrasing below and in Pass 3.
- People-centric: describe what the reader can do, not what the feature is. Do: `Find people by searching for their name or email.` Don't: `The search tool can search by name and email.`

> - Positive phrasing; do not blame the user. Do: `Enter alphanumeric characters only.` / `Your password is incorrect.` Don't: `Don't enter special characters.` / `You entered an incorrect password.`

Reviewer: line 120 promises sentence quality checks but where is it?
and where did sentence length go? did you remove that on purpose?

```suggestion
- Positive phrasing; do not blame the user. Do: `Enter alphanumeric characters only.` / `Your password is incorrect.` Don't: `Don't enter special characters.` / `You entered an incorrect password.`
- **Sentence length**: ~20 words average. Split longer sentences or convert to lists/tables.
```

Author: **Two parts — first is a real gap; second suggestion has no source and shouldn't be added as-is.**

1. **"sentence quality" promised but missing** — confirmed gap. Pass 2's intro (line 120) lists "sentence quality," but no such rule appears in Pass 2. This is an over-promise in the intro.
2. **"Sentence length: ~20 words average"** — I traced this to both sources and it appears in **neither**. The Help Portal *Global English* chapter covers "sentences must sound natural," common words, syntactic cues, articles, "that", phrasal verbs — but **no sentence-length / 20-word rule**. So adding it would be inventing a rule with no upstream basis.

**Recommended fix:** drop "sentence quality" from the line-120 intro (align the promise to what the sources actually provide), rather than fabricate a length rule. If we *want* a length guideline, that's a net-new Kyma convention to decide deliberately — flagging separately. The **positive-phrasing** suggestion in your block *is* source-backed (Help *Message* guidance / Community *Error messages do not blame the user*) and is already at line 134, so that part's fine.

Reviewer: What about this:

> Use simple syntax. Limit sentence length to 20 words.

Source: SAP Style Guide for Technical Communication, Style and Wording: Practical Issues and Principles https://help.sap.com/docs/SNG_FOR_UA/e33c591ae4494a659a3f5f983c9d1161/0e1ae688897b4bc185aa17002bae8ef4.html?locale=en-US&state=DRAFT

> ### Recommendations (override — preserve advisory tone, no first person)

Reviewer: ```suggestion
### Recommendations (Override — Preserve Advisory Tone, No First Person)
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

When content is advisory (optional/best-practice), keep it advisory — do not convert to a bare imperative (that changes "suggested" to "required"). The should→must/can rule and the imperative-mood rule do **not** apply here.
- Approved patterns: `It is recommended to {action}.` (accepted passive exception), `Consider {action}.`, `As a best practice, {action}.`, or a **Recommendation** panel.
- Flag & rewrite: `We recommend ...` → use an approved pattern. `It is not recommended ...` → state positively what to do instead.

> ### Grammar and clarity

Reviewer: ```suggestion
### Grammar and Clarity
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

- Articles: "the" for a specific thing, "a"/"an" for non-specific; choose a/an by pronunciation (`an SAP solution`). Include articles to remove ambiguity.
- Use "that" to introduce object clauses and in restrictive relative clauses even when omittable. Do: `Define the information that you require.`
- Restrictive clauses: "that", no commas. Nonrestrictive: "which"/"who", with commas.
- Keep phrasal verbs together (`Fill out the form.`). Prefer a relative clause over a post-noun participle (`the template that was used`, not `the template used`).
- Don't use causative "have"/"get" ("get a page created") — name the actor. Avoid ambiguous "since" (because vs. from-the-time). Avoid dangling modifiers. Clarify what "and"/"or" join. Split infinitives only for contrast.
- Collective nouns take a singular verb (`The data is saved`, `SAP is launching`).
- Serial comma (see Pass 1). Use a colon to introduce a list, a semicolon between independent clauses — but prefer neither. Avoid parentheses; prefer a list or restructured sentence.

> - Contractions: acceptable in moderation; write out in full for serious warnings (`do not`, `cannot` — never `can not`); no unusual/compound contractions; watch its/it's.

Reviewer: I'd argue SAP **encourages** [using contractions](https://help.sap.com/docs/SNG_FOR_UA/e33c591ae4494a659a3f5f983c9d1161/64c7824e02f642a4a2842612fb0cd18d.html?locale=en-US&state=DRAFT&version=PROD) (in moderation) because it makes "content sound more conversational, an important aspect of the people-centric approach".

Note that our CCM's built-in checker (Acrolinx) wants us to use **more** contractions in our docs, see:

<img width="182" height="88" alt="Image" src="https://github.com/user-attachments/assets/b286de32-178e-4c3a-a5f6-2a47abeb6139" /> 

<img width="174" height="87" alt="Image" src="https://github.com/user-attachments/assets/b1d08ad5-867c-4c46-992e-19def94891e1" /> 

<img width="177" height="80" alt="Image" src="https://github.com/user-attachments/assets/f91cb896-0cfc-446b-8589-b44e43d0b353" /> 

<img width="176" height="69" alt="Image" src="https://github.com/user-attachments/assets/5e91275f-334a-4654-b161-94cfd9c5b6f1" /> 

<img width="184" height="100" alt="Image" src="https://github.com/user-attachments/assets/405c8710-05cd-4ac5-914b-b0b8e2c34cc9" /> 

<img width="168" height="75" alt="Image" src="https://github.com/user-attachments/assets/eb4a17da-9844-4209-9138-3c76f427d2d0" />

Author: **You're right, and the Help Portal source backs you — the guide understates it.** This is not a self-violation but a case where the current wording is weaker than the source.

- **Help Portal** source, chapter *Contractions → Use common contractions in moderation*:
  > **Rule:** Use common contractions to sound natural (test by reading aloud); if in doubt, write the form out in full.
  > **Do:** "When you sign in to the system, you **don't** need to provide your account information."
  > **Don't:** "...you **do not** need..." *(robotic)*

  The source actively *encourages* contractions (labels the spelled-out form "robotic") and ships a whole table of accepted ones (`aren't`, `can't`, `doesn't`, `isn't`, `it's`, `you're`, ...). It only bars them in **serious warnings** ("do not"/"cannot") and bans compound/informal ones.

So your Acrolinx/CCM point aligns with the guide's own upstream source. The final line ("acceptable in moderation") is faithful but too terse. **Suggested fix (our side):** reword to reflect the source's positive framing — "Use common contractions to sound natural; write out in full only in serious warnings (`do not`, `cannot`) and avoid compound/informal ones." Flagging for us to revise.

Reviewer: agree

### US English

- American spelling: `color`, `behavior`, `organize`, `center`, `catalog`, `defense`, `license`, `fulfillment`, `program`. Not British (`colour`, `organise`, `centre`, `-ogue`, `-our`).

> - American terms over British: `fill out` (not fill in), `for more information about` (not on), `parentheses`/`curly brackets`/`square brackets`, `period`, `exclamation point`, `vacation`, `expiration date`.

Reviewer: Contradiction:

here, you say use "fill out" instead of "fill in".
in line 232, you say use "fill in" instead of "complete".

Author: **Confirmed — same contradiction as line 232, viewed from the Pass 2 side.**

Here the guide is **faithful to the Help Portal source**:

- *US vs British Terms*: `fill out` is the correct American form (not British `fill in`).
- *Global English → Keep phrasal verbs together*: **Do:** "Fill out this application form."

The conflict is with the **Community** *Preferred terminology table* row `fill in` → `complete` (surfaces at line 232). The Community row's real intent is to ban `complete`, not to prescribe British `fill in`. Recommend this line stays (`fill out`) and the line-232 row is rewritten to `fill out` → avoid `complete`, `fill in`. See the line-232 thread.

Reviewer: yes, let's follow the Help Portals source (and also, community guidance should probably be updated)

- Hyphenate American-style (less than British): hyphen mainly to prevent misreading. Hyphenate an adjective **before** a noun only (`well-known actor` vs. `is well known`); no hyphen when an adverb modifies it (`highly configurable`). Prefixes usually closed (`autogenerated`, `nonnegotiable`, `preassembled`) except all-/cross-/self-/high-/well-/ex-/full-.

### Global English — common words and clear structure

Prefer common, everyday words; avoid telegraphic style (keep conjunctions, articles, prepositions). Sentences must sound natural and be unambiguous.

> **Plain-word substitutions (use → avoid):**

Reviewer: line 43 says "Do not enclose trailing punctuation inside formatting"

```suggestion
**Plain-word substitutions (use → avoid)**
```

Author: ---

**Correction (please disregard the "accepting" above).** This flags a formatting rule applied to the styleguide's *own* text. But `kyma-styleguide.md` is **not** documentation subject to these rules — it is an **instruction document read by an AI agent** that reviews *other* docs. The rule is for the *target* content the agent checks, not for the styleguide's prose.

So the styleguide "not obeying its own rule" here is expected and harmless — it is not self-applied. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed self-violation**, same as line 20. The bold lead-in encloses its trailing colon inside the formatting, against the guide's own rule (line 43): "Do not enclose trailing punctuation inside formatting — translation tools segment on them." Accepting `**Plain-word substitutions (use → avoid):**` → colon outside the bold.

| Use | Avoid |
| --- | --- |
| agree | concur |
| although | albeit |
| before | prior to |
| start, begin | commence |
| try | endeavor |
| up to now | heretofore |
| use | utilize |
| get | obtain |
| happen | occur |
| harmful | deleterious |
| can | be able to |
| upgrade | perform an upgrade |
| explain | provide an explanation |
| is transparent | is characterized by transparency |
| that is | i.e. |
| for example, such as | e.g. |
| and so on | etc. |
| note | n.b. |
| and others | et al. |
| versus | vs. |
| compare | cf. |

- Do not use Latin abbreviations in body text (space-restricted UI may keep `e.g.`). Cross-ref: preferred Kyma terms are in Pass 3.

---

> ## Pass 3 — Terminology and naming

Reviewer: ```suggestion
## Pass 3 — Terminology and Naming
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

What to call things — product names, Kyma terminology, resource naming, abbreviations.

> ### Product naming (SAP BTP, Kyma runtime)

Reviewer: ```suggestion
### Product Naming (SAP BTP, Kyma Runtime)
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

- Do **not** use: `SAP BTP Kyma`, `SAP Kyma`, `SAP BTP, KR`, or any invented abbreviation.
- Space-limited web/menu/tile contexts only: `SAP BTP Kyma runtime` or `Kyma runtime` (comma/`SAP BTP` may drop when context is clear). Cockpit/service-catalog tiles use the short name `Kyma Runtime`.
- Body text may use natural phrasing like `the Kyma runtime for SAP BTP`. Spell out "Business Technology Platform" early in a document to gloss the abbreviation.
- Dashboard: **Kyma dashboard** (lowercase "dashboard" except in titles). Do not precede with `SAP`/`SAP BTP` (`SAP Kyma dashboard` is wrong). First mention may use `Kyma dashboard for SAP BTP, Kyma runtime`; later just `dashboard`.
- General SAP offering names: use the approved name exactly — no plurals (rephrase: `SAP CRM systems`), no possessives (`SAP HANA benefits`), no trademark symbols, no italics/quotes, no invented abbreviations. Title case in titles even if the body form has lowercase words.
- Descriptors/articles for SAP offerings: at first occurrence use the approved descriptor (`the SAP Business One solution`); afterward use the name stand-alone with no "the" and no descriptor, or a descriptor with "this".

> ### No article before Kyma product/component names

Reviewer: line 112 says use "or" instead of slash

```suggestion
### No Article Before Kyma Product or Component Names
```

Author: ---

**Correction (please disregard the "accepting" above).** This flags a formatting rule applied to the styleguide's *own* text. But `kyma-styleguide.md` is **not** documentation subject to these rules — it is an **instruction document read by an AI agent** that reviews *other* docs. The rule is for the *target* content the agent checks, not for the styleguide's prose.

So the styleguide "not obeying its own rule" here is expected and harmless — it is not self-applied. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed self-violation.** The heading uses `product/component` as slash-for-"or", against a rule backed by both sources.

- **Help Portal** source, *Slashes → Do not use a slash as shorthand for "or"*:
  > **Rule:** Do not use a forward slash as a shorthand for "or" or similar meanings; replace it with "or"...
  > **Don't:** You can change/delete a bill of material at any time.
- **Community** source, *Preferred terminology table*: `or` → avoid `/ (slash)`.

Accepting `product/component` → `product or component`. (For reference, the slashes elsewhere in the guide — `key-value` alternatives, `plan/actual`, `SAP S/4HANA` — are the Help source's explicit unit-modifier and approved-name exceptions, so those stay.)

Do not use an article before the product or a component name. Do: `Kyma is awesome.`, `Application Connector`. Don't: `The Kyma`, `the Application Connector`. (Note: this Kyma rule coexists with the SAP first-occurrence-descriptor rule for SAP *offering* names above; for Kyma component names, no article.)

> ### Kubernetes / resource naming reference

Reviewer: ```suggestion
### Kubernetes / Resource Naming Reference
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

- Always capitalize **Kubernetes**; never `kubernetes` or `k8s`.

> - CamelCase for Kubernetes and custom resources; "s" for plurals; spaces in titles/navigation. Format as code (`` `APIRule` ``) only when referring to the code specifically. Cross-ref: body-text formatting rule in Pass 1.

Reviewer: The boundary between "referring to the resource in the code specifically" and "referring to the resource conceptually" is subjective. A reviewing AI may struggle to determine intent. Consider adding 2–3 more disambiguating examples (e.g., "When you apply the APIRule manifest..." vs. "API Gateway operates on APIRule custom resources.").

Author: **Fair point — additive suggestion, and the disambiguation itself is source-backed.** Not a violation; you're asking for more examples to make the code-vs-conceptual boundary decidable.

The underlying rule is faithful to **Community** *Format resource names as code when referring to code*:
> **Rule:** Format Kubernetes and custom resource names as code (`` `APIRule` ``) only when referring to them in code specifically; otherwise use plain CamelCase.

The boundary is genuinely subjective, as you say. Adding 2–3 contrasting examples (e.g. "When you apply the `APIRule` manifest..." vs. "API Gateway operates on APIRule custom resources.") is a reasonable clarity improvement with no source conflict. Accepting as an enhancement.

- Do **not** capitalize `namespace`.
- Kubernetes resource names (CamelCase): `ConfigMap`, `CronJob`, `CustomResourceDefinition (CRD)`, `Deployment`, `Function`, `Ingress`, `Node`, `Pod`, `PodPreset`, `ProwJob`, `Secret`, `Service`, `ServiceBinding`, `ServiceClass`, `ServiceInstance`.
- Title case for Kyma component names: `Application Connector`, `API Gateway Controller`.

> ### Kyma preferred terminology (use → avoid)

Reviewer: ```suggestion
### Kyma Preferred Terminology (Use → Avoid)
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

| Use | Avoid | Note |
| --- | --- | --- |
| ID | id | |
| and | + , & | |
| backend | back end, back-end | |
| frontend | front end, front-end | |
| micro frontend | microfrontend, micro front-end | |
| connect, connection | integrate, integration | |
| custom resource | Custom Resource, CustomResource | |
| document | doc | |
| email | e-mail | |

> | fill in | complete | |

Reviewer: Contradiction:

in line 156, you say use "fill out" instead of "fill in".
here, you say use "fill in" instead of "complete".

Author: **Confirmed contradiction — and it traces to a genuine source conflict, not a generation slip.**

The two sources disagree:

- **Help Portal** source, chapter *US vs British Terms*, mandates the American form:
  > `fill out [a form]` (American, correct) — not `fill in [a form]` (British)

  Reinforced in *Global English → Keep phrasal verbs together*: **Do:** "Fill out this application form."

- **Community** source, *Preferred terminology table*, has the opposite row:
  > | Use | Avoid |
  > | `fill in` | `complete` |

So the final guide faithfully carried **both** through: line 156 (from Help, US English → `fill out`) and line 232 (from Community table → `fill in`). The assembly pass should have reconciled these and didn't.

**Resolution needed (our side):** pick one. The Help Portal is the rendered source of truth and is explicit that `fill out` is the American form; the Community `fill in`→`complete` row only intends to ban `complete`, not to prescribe British `fill in`. Recommend keeping `fill out` (Pass 2) and rewriting the Community row as `fill out` → avoid `complete`, `fill in`. Paired with the line-156 comment.

Reviewer: Help Portal source wins, hands down. 
For Kyma/K8s-specific knowledge, the community source might be authoritative, but for such generic language guidance, the Help Portal source beats community.

| for example, such as | e.g. | mid-sentence, prefer "such as" |
| cloud-native (adj.) | cloud native | adjective compounds usually hyphenated |
| application | app | "Application" = external solution via Application Connector; "application" = a microservice on Kyma |
| the following | below, this, the described | |
| the previous, earlier | above, this, the described | |
| Infrastructure Provider, IaaS Provider | hyperscaler, Cloud Provider | |

- Do not use transitional-phase words (`currently`, `now`) or make forward-looking promises; document only what already exists.
- Short command-line arguments where possible (`-n`, not `--namespace`); explain context if ambiguous. Note: `kubectl -n` = `--namespace`, but `helm -n` = `--name`, not `--namespace`.

### Abbreviations

- Prefer no abbreviation; introduce one only if more common than the full form or an industry standard, and never coin on the fly. At first occurrence give `full form (ABBR)`, then use the abbreviation; don't alternate. No abbreviations in titles.
- Plurals without apostrophe (`IDs`, `BAPIs`). No CamelCase in abbreviations of translatable words (CamelCase is reserved for technical names). Use "and", not `&`/`+`.

> ### Country/region

Reviewer: ```suggestion
### Country/Region
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

Do not use "country" alone — use `country/region` or `country or region`, or rephrase. Do: `Select the relevant countries/regions.`

> ### Software objects and entities

Reviewer: ```suggestion
### Software Objects and Entities
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

- Real-life entity or where "this is software" is irrelevant: write like any other word — no title case, no formatting. Do: `A strategic purchaser creates a contract.` Don't: `As a Strategic Purchaser ...`.
- Software object (role, API, function module, etc.) at first occurrence: `*Title in Title Case* (\`technical name\`) descriptor`. Later, use the title alone or the technical name. Omit elements that don't apply.
- Software entity behaving as a proper noun (tool/engine, one exists, no plural): title case, no formatting, use an article, do not prefix with `SAP`. Ex: `The BAdI Builder`.
- AI: make it obvious users interact with AI; use the AI's name, avoid pronouns where possible.
- Cross-ref: topic-type naming (concept/task/troubleshooting/CR titles) is in Document structure.

---

> ## Pass 4 — Reconciliation sweep

Reviewer: ```suggestion
## Pass 4 — Reconciliation Sweep
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

> 

Reviewer: Passes 1-3 have concrete rules. Pass 4 says "verify no pass introduced a new violation" but gives no decision framework for conflicts. Which pass wins when fixing a Pass 3 term breaks Pass 2 sentence flow?

Author: **Confirmed gap — assembly-layer, no source backing for a specific resolution.** Neither source defines pass precedence (both are flat rule sets), so there's no source to cite; this is a design decision for our assembly prompt.

You're right that Pass 4 detects conflicts but gives no tie-breaker. The guide *does* encode local overrides where sources conflict (e.g. Kyma gerund rule overrides SAP's gerund default at line 68; Help Portal panel syntax overrides Community VitePress alerts at line 46). But there's no *general* precedence rule. **Flagging for us to add** an explicit ordering to Pass 4 — candidate: terminology/naming (Pass 3) and formatting overrides win on the specific token; sentence-flow (Pass 2) yields locally and gets rephrased around the fix. To be designed.

Re-read the changes from Passes 1–3 together and verify the document as a whole:

- Verify no pass introduced a new violation of another pass (e.g. a terminology fix in Pass 3 that broke sentence flow, or a formatting change that de-capitalized a proper noun).
- Check consistency across the full document: one term per meaning; consistent capitalization per text type; consistent list/table punctuation; consistent term-definition style; heading style uniform.
- Verify the document conforms to its **topic type** (concept, task, troubleshooting, custom resource) per Document structure below — correct title style, required sections, and shape.
- Confirm advisory panels are within moderation limits and use valid Help Portal syntax.
- Where a situation matched no rule, confirm the best-practice judgment applied is consistent with the rest of the document.

---

> ## Document structure and conventions

Reviewer: ```suggestion
## Document Structure and Conventions
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

Whole-file / topic-shape rules. Not a numbered pass. Each rule is cross-referenced from its owning pass.

> ### Topic types (cross-ref: Pass 3 for titling, Pass 1 for heading formulation)

Reviewer: ```suggestion
### Topic Types (Cross-Ref: Pass 3 for Titling, Pass 1 for Heading Formulation)
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

Kyma is topic-based: one file per topic, self-contained, linking out as needed.

- **Concept** (answers "what is"): nominal title (`Security`, `Security Concept`). Background only — no instructions, no reference tables/lists; may link to task/reference topics.

> - **Task** (a task the reader accomplishes): imperative-verb title naming the task, not the feature, not a gerund, not "How to...". Do: `Define resource consumption`. Don't: `Selecting a profile`, `How to select...`. Structure: intro paragraph ("why do this?") → prerequisites (if any) → numbered steps → expected result. Aim for 5–9 steps; split longer.

Reviewer: ```suggestion
- **Task** (a task the reader accomplishes): imperative-verb title naming the task, not the feature, not a gerund, not "How to...". Do: `Define resource consumption`. Don't: `Selecting a profile`, `How to select...`. Structure: intro paragraph ("why do this?") → prerequisites (if any) → numbered steps → expected result. Aim for 5–9 steps; split longer tasks.
```

Author: **Suggestion is source-faithful — accepting.** The Task-topic definition traces to **Community** *Task topic title describes the task, not the feature* and *Task topic structure*:
> title describes the task (imperative verb), not the feature, not a gerund, not "How to..."; structure: intro → prerequisites → numbered steps → expected result.

Note one cross-source tension worth being aware of: the **Help Portal** source (*Titles and Headings → Procedural titles: gerund*) actually *mandates* gerunds for procedural titles ("Setting Up Authorizations"). The Kyma convention (Community) overrides that, which the guide already flags at line 68. Your reworded block keeps the imperative-verb Kyma form, so it's consistent with the override. Accepting.

Reviewer: The imperative vs gerund discussion is a hot topic in UA and the situation more complex than we can reasonably cover in an automatic reviewer. Bottom line - if at least a doc is consistent in itself and with its surrounding docs, we're on a good path.

- **Troubleshooting:** title names the symptom/error the reader sees, not the cause. Do: `Cannot access ...`. Don't: `Incompatible version`. Structure: three headings — **Condition**, **Cause**, **Solution** (numbered list for sequential steps; bullets or sub-headings for several equally valid solutions).
- **Custom resource (CR):** file named `{RESOURCE_NAME}.md`; title is the CR name in CamelCase (`LogPipeline`, `Function`).

> ### Shell commands and code blocks (cross-ref: Pass 1 formatting)

Reviewer: ```suggestion
### Shell Commands and Code Blocks (Cross-Ref: Pass 1 Formatting)
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case per **Community** *Write headings in title case* and **Help Portal** *Capitalization*; restated at line 16. Accepting.

Section content is faithful to **Community** *Runnable shell commands have no prompt prefix*, *Command-plus-output transcripts prefix only command lines*, and *Mark omitted code with an ellipsis* — the guide's two code-block examples (bare `kubectl` vs. `$ `-prefixed transcript) come straight from those rules.

- **Runnable command** (meant to be copied): show the command alone — no prompt prefix, no `$`/`#`/path/hostname. Use environment variables instead of hardcoded values; omit `export` unless the value is noteworthy.
  ```
  kubectl -n $NAMESPACE get pods -l app=$APP_LABEL
  ```
- **Command-plus-output transcript:** prefix only command (input) lines with `$ `; leave output unprefixed; omit path/hostname.
  ```
  $ kubectl -n kyma-system get pods
  NAME                          READY   STATUS    RESTARTS   AGE
  istiod-7c8b9f4d4f-abcde       1/1     Running   0          12d
  ```
- Mark omitted parts of a snippet or response with `...`.

> ### Links (cross-ref: Pass 2 for link-text wording)

Reviewer: ```suggestion
### Links (Cross-Ref: Pass 2 for Link-Text Wording)
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.

- Relative links within the same repository; absolute links for other repos and external sources.
- Descriptive link text; never vague ("this", "that", "here") and don't link whole phrases. Prefer the cross-reference pattern `For more information, see [target].` Keep links syntactically separate from descriptive sentences (in a cross-reference formulation, after an introductory colon, or in a closing "Related Information" section).
- Don't over-link: skip content that is peripheral, well-known to the audience, or easily searched. Keep to a manageable number per topic.
- Link to a heading with `#{name-of-the-heading}` appended to the filepath. Reference assets (YAML/JSON/SVG/PNG/JPG) in the topic's `assets` folder via a GitHub relative link.
- When mentioning a configuration file, consider linking to it instead of naming it; use the name without the format extension.

> ### Advisory information moderation (cross-ref: Pass 1 panel syntax/types)

Reviewer: ```suggestion
### Advisory Information Moderation (Cross-Ref: Pass 1 Panel Syntax/Types)
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case per **Community** *Write headings in title case* and **Help Portal** *Capitalization*; restated at line 16. Accepting.

Section content is faithful to **Help Portal** *Notes, Tips, Cautions, and the Like → Use advisory information in moderation*:
> **Rule:** Use all types of advisory information in moderation: as a general rule, no more than 2 items of advisory information per topic, and do not place them directly after one another... a simple "Note that" sentence in body text is sometimes sufficient.

The guide's "~2 per topic," "not directly one after another," and "a plain 'Note that...' sentence is sometimes enough" all map to this rule (plus *Provide repeated advisory information centrally*).

Use advisory panels in moderation — as a rule no more than ~2 per topic, and not directly one after another. Overuse dilutes them. A plain "Note that ..." sentence in body text is sometimes enough. Provide repeated advisory info centrally.

> ### Examples and sample data

Reviewer: ```suggestion
### Examples and Sample Data
```

Author: ---

**Correction (please disregard the "accepting" above).** This comment — and my reply — mis-frame the document. `kyma-styleguide.md` is **not** SAP Help Portal content; it is an **instruction document whose audience is an AI agent** reviewing *other* people's docs. Its rules govern the *target* documentation, not the styleguide's own prose.

Heading case in an agent-facing instruction file does not affect how the agent applies the rule, so this is not a defect. No change needed on this ground; parking as "not self-applied."

*(Original reply retained below for history.)*

---
**Confirmed — correct and source-backed.** Title case for headings is faithful to both sources:

- **Community** source, *Write headings in title case*:
  > **Rule:** Write headings in title case. **Do:** **Expose a Service**
- **Help Portal** source, chapter *Capitalization* (sentence style vs. title case): titles/headings take title case.

The guide restates this at line 16 ("Titles, headings... title case"), so applying it to this heading is self-consistent. Accepting.
